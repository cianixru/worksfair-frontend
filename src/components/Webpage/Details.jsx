import React from 'react';
import PropTypes from 'prop-types';

const DetailItems = ({
  details,
}) => {
  return (
    <ul className="columns">
      { details.length > 0
        && details.map(detail => (
          <div
            key={detail.title}
            className="column is-4">
            <li>
              <div className="columns is-desktop">
                <div className="column">
                  <h4
                    className="title is-5 detail-title has-text-grey"
                  >
                    <span
                      style={{
                        maxHeight: '7px',
                        marginLeft: '30px',
                        marginRight: '10px',
                        padding: '0px 3px',
                        backgroundColor: `${
                          detail && detail.colour
                        }`,
                      }}
                    />
                    {detail.title}
                  </h4>
                  <div
                    className="detail-content has-text-grey"
                    style={{
                      borderLeft: '30px solid',
                      borderBottom: '15px solid',
                      borderColor: `${
                        detail && detail.colour
                      }`,
                    }}
                  >
                    <p>{detail.description}</p>
                  </div>
                </div>
              </div>
            </li>
          </div>
        ))
      }
    </ul>
  );
};

DetailItems.propTypes = {
  details: PropTypes.array,
};

export default DetailItems;
