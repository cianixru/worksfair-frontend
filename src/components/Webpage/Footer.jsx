import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const WebpageFooter = ({ title }) => (
  <footer className="columns footer has-text-centered">
    <div className="column">
      <div className="copyrights">
        <h4 className="subtitle is-5" >
          Copyright <b>{title}</b> {new Date().getFullYear()}
        </h4>
        <h6 className="subtitle is-size-7" >Powered by
          <Link
            to="/"
            className="has-text-grey" > Worksfair.com</Link>
          . All Rights Reserved.
        </h6>
      </div>
    </div>
  </footer>
);

WebpageFooter.propTypes = {
  title: PropTypes.string,
};

export default WebpageFooter;
