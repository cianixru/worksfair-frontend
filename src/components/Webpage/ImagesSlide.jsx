/* eslint-disable jsx-a11y/anchor-has-content */
import React from 'react';

const ImagesSlide = ({ webpage }) => {
  return (
    <div className="listing-slider mfp-gallery-container margin-bottom-0">
      {
        webpage.images
      }
      <a href="images/single-listing-01.jpg"
        backgroundImage="images/single-listing-01.jpg"
        className="item mfp-gallery"
        title="Title 1" />
    </div>
  );
};

ImagesSlide.propTypes = {
  webpage: PropTypes.object,
};

export default ImagesSlide;
