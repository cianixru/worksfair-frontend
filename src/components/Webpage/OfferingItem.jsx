import React from 'react';
import PropTypes from 'prop-types';

import { addNairaSign } from '../utils/helpers';

const OfferingItem = ({
  offerings,
  colour,
}) => {
  return (
    <ul className="columns">
      { offerings.length > 0
        && offerings.map(offering => (
          <div
            key={offering.title}
            className="column is-6">
            <li
              className="box"
            >
              <div className="columns is-desktop">
                <div className="column">
                  <figure className="image is-180x180">
                    <img
                      src={offering.image}
                      alt=""
                      className="offering-image"
                    />
                  </figure>
                </div>
                <div className="column">
                  <h4 className="title is-5">
                    {offering.title}
                  </h4>
                  <p>{offering.description}</p>
                  <div className={`tag ${colour} is-large`}>
                    {addNairaSign(offering.price)}
                  </div>
                </div>
              </div>
            </li>
          </div>
        ))
      }
    </ul>
  );
};

OfferingItem.propTypes = {
  offerings: PropTypes.array,
};

export default OfferingItem;
