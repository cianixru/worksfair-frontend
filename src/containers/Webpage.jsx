import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import OwnerCard from '../components/Webpage/OwnerCard';
import TitleBar from '../components/Webpage/TitleBar';

class WebpageContainer extends Component {
  async componentDidMount() {
    try {
      const { match, getWebpage } = this.props;
      await getWebpage(match.params.subDomainName);
    } catch (error) {
      console.log(error);
    }
  }

  render() {
    const { webpage } = this.props;
    return (
      webpage
      && <div>
        <section
          className={`hero is-medium webpage-hero ${webpage.colour}`}>
          <div className="hero-body">
            <div className="container">
              <h1 className="title">
                { `Welcome to ${webpage.title}` }
              </h1>
              <h2 className="subtitle">
                {webpage.description }
              </h2>
            </div>
          </div>
        </section>
        <div className="container">
          <div className="columns is-desktop">
            <div className="column is-three-quarters">
              <TitleBar webpage={webpage} />
            </div>
            <div className="column is-one-quarter">
              <OwnerCard owner={webpage && webpage.owner}/>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

WebpageContainer.propTypes = {
  getWebpage: PropTypes.func,
  webpage: PropTypes.object,
  match: PropTypes.object,
};

WebpageContainer.defaultProps = {
  webpage: {},
};

const mapStateToProps = ({ webpage, loader, }) => ({
  webpage: webpage.webpage,
  isLoading: loader.isLoading,
});

export default connect(
  mapStateToProps,
  null,
)(WebpageContainer);
