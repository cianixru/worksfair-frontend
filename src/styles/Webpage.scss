.webnav {
    position: relative;
    z-index: 999;
    margin-bottom: 5px;
    box-shadow: 0 0 8px 0 rgba(0, 0, 0, .12);
    font-size: 16px;
}

.webpage-title {
    font-weight: 700;
    margin: 0 50px;
    font-size: 24px;
}

.colour-holder {
    padding: 0 12px;
    margin-right: 5px;
}

.webpage-hero {
    margin-bottom: 50px;
}

.webpage-address {
    font-size: 17px;
    color: #888;
    margin-top: 0;
    display: inline-block
}

.webpage-address i {
    margin-right: 5px
}

.webpage-sections {
    margin: 30px 0px;
}

.webpage-sections span a:hover {
    color: #666
}

.webpage-sections h1,
.webpage-sections h2 {
    font-size: 28px;
    line-height: 36px;
    margin: 1px 0
}


.verified-icon {
    display: inline-block;
    width: 20px;
    height: 20px;
    font-size: 22px;
    color: #00ace6;
    font-style: normal;
    font-weight: 500;
    line-height: 20px;
    text-align: center;
    position: relative;
    top: 3px;
    margin-left: 2px;
    margin-right: 10px;
}

.verified-icon:before {
    content: "\f058";
    font-family: fontawesome;
    position: relative;
    z-index: 11;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    display: block
}

.verified-icon:after {
    position: relative;
    z-index: 9;
    width: 17px;
    height: 17px;
    border-radius: 50%;
    display: block;
    content: "";
    position: absolute;
    top: 1px;
    left: 1px;
    background-color: #fff
}

.verified-badge {
    border-radius: 4px;
    color: #fff;
    text-align: center;
    padding: 12px 20px;
    font-size: 17px;
    font-weight: 500;
    cursor: help;
    position: relative;
    transition: .3s;
    display: block;
    margin-top: 30px;
}

.verified-badge i {
    font-size: 22px;
    padding: 0 4px 0 0;
    position: relative;
    top: 3px;
}

.verified-badge .tip-content {
    transform: translate(-50%, 100%);
    -ms-transform: translate(-50%, 100%);
    padding: 20px;
    border-radius: 0 0 4px 4px;
    max-width: auto;
    top: auto;
    bottom: 0
}

.verified-badge:hover {
    border-radius: 4px 4px 0 0
}

.webpage-slider .item {
    position: relative;
    background-size: cover;
    background-position: 50%;
    height: 40vh;
    min-height: 400px
}

.webpage-slider,
.webpage-slider .item {
    height: 40vh;
    min-height: 400px
}

a.mfp-gallery {
    cursor: zoom-in
}

.webpage-slider {
    background-color: #fff
}

.webpage-slider .slick-next:hover,
.webpage-slider .slick-prev:hover,
.webpage-slider .slick-next,
.webpage-slider .slick-prev {
    background-color: rgba(20, 20, 20, .45);
    color: #fff;
    height: 60px;
    width: 60px;
    border-radius: 50%;
    text-align: center;
    transition: .4s
}

.webpage-slider .slick-next:hover,
.webpage-slider .slick-next {
    right: 15%;
    transform: translate(50px, -50%)
}

.webpage-slider .slick-prev:hover,
.webpage-slider .slick-prev {
    left: 15%;
    transform: translate(-50px, -50%)
}

.webpage-slider:hover .slick-next {
    transform: translate(60px, -50%)
}

.webpage-slider:hover .slick-prev {
    transform: translate(-60px, -50%)
}

.webpage-slider .slick-next:hover:before,
.webpage-slider .slick-prev:hover:before,
.webpage-slider .slick-prev::before,
.webpage-slider .slick-next::before {
    font-size: 20px;
    line-height: 60px;
    position: relative;
    color: #fff
}

.webpage-slider .slick-prev::before {
    left: -1px
}

.webpage-slider .slick-next::before {
    left: 1px
}

.webpage-slider .slick-next:hover {
    transform: translate(64px, -50%)
}

.webpage-slider .slick-prev:hover {
    transform: translate(-64px, -50%)
}

.webpage-slider .item.slick-slide:before {
    position: absolute;
    height: 100%;
    width: 100%;
    display: block;
    content: "";
    background: #222;
    z-index: 100;
    visibility: visible;
    opacity: .1;
    transition: .3s ease-in-out
}

.webpage-slider .item.slick-slide:hover:before,
.webpage-slider .slick-next,
.webpage-slider .slick-prev {
    opacity: 0;
    visibility: hidden
}

.webpage-slider:hover .slick-next,
.webpage-slider:hover .slick-prev {
    opacity: 1;
    visibility: visible
}

.webpage-slider .slick-next:hover,
.webpage-slider .slick-prev:hover {
    background-color: #66676b;
    color: #fff
}

.webpage-offerings {
    padding: 0.75em;
}

.titles {
    margin-top: 70px;
}

.price {
    font-size: 1.15rem;
    font-style: italic;
    font-weight: 400
}
