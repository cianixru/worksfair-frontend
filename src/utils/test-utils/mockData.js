// eslint-disable-next-line import/prefer-default-export
export const user = {
  username: 'Theookafor2',
  first_name: 'eod',
  last_name: 'okafor',
  email: 'theodorehope@andela.com',
  facebook: null,
  instagram: null,
  twitter: null,
  headline: null,
  current_location: null,
  role: 'admin',
  confirmed_account: false,
  verified_account: false,
  reset_token: null,
  image_url: null,
  created_at: '2019-03-10T08:48:42.942532Z',
  modified_at: '2019-03-10T08:48:42.942544Z',
  token: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9',
};

export const webpage = {
  owner: {
    id: 1,
    username: 'theowa',
    first_name: 'theo',
    last_name: 'IO',
    email: 'theo.iowa@yahoo.com',
    facebook: null,
    instagram: null,
    twitter: null,
    headline: null,
    current_location: null,
    role: 'admin',
    confirmed_account: false,
    verified_account: false,
    reset_token: null,
    image_url: 'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png',
    created_at: '2019-04-25T09:07:08.209916Z',
    modified_at: '2019-04-25T09:07:08.209930Z'
  },
  title: 'Happiness',
  sub_domain_name: 'happiness',
  description: 'Happiness truly lives here. We just happy',
  keywords: 'passion, software, engineering, technology, worksfair',
  country: 'Nigeria',
  state: 'Lagos',
  city: 'Okota',
  area: 'Omagba Phase 1, Phase 2, Abatete',
  address: 'No 7 Obinna Uzor Street, Canal Estate',
  phone: null,
  email: null,
  website: null,
  featured_images: [
    'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png',
    'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png'
  ],
  is_active: true,
  logo: null,
  colour: 'is-dark',
  created_at: '2019-04-28T18:39:25.174272Z',
  updated_at: '2019-05-01T12:18:49.669633Z'
};
