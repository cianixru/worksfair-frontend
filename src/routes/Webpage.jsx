import React, { Component } from 'react';
import { withRouter, Switch, Route } from 'react-router-dom';
import PropTypes from 'prop-types';

import Webpage from '../containers/Webpage';

class WebpageRoute extends Component {
  render() {
    const { getWebpage } = this.props;
    return (
      <Switch>
        <Route exact path="/webpage/:subDomainName"
          // eslint-disable-next-line react/jsx-no-bind
          render={(props) => {
            return (<Webpage {...props} getWebpage={getWebpage} />);
          }}
        />
      </Switch>
    );
  }
}

WebpageRoute.propTypes = {
  getWebpage: PropTypes.func,
};

export default withRouter(WebpageRoute);
