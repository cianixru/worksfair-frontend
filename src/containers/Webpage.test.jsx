import React from 'react';
import { Provider } from 'react-redux';
import { render, cleanup } from 'react-testing-library';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';

import Webpage from './Webpage';
import { webpage } from '../utils/test-utils/mockData';

afterEach(cleanup);

const middlewares = [thunk];
const store = configureMockStore(middlewares);
const mockStore = store({
  auth: { currentUser: {} },
  webpage: { webpage, },
  loader: { isLoading: false, },
});

describe('Webpage.jsx', () => {
  const props = {
    match: {
      params: { subDomainName: 'name' },
    },
    getWebpage: jest.fn(),
  };

  test('Should display the Webpage Component', () => {
    const component = render(
      <Provider store={mockStore}>
        <Router>
          <Webpage {...props} />
        </Router>
      </Provider>
    );
    expect(component).toBeDefined();
    expect(component.getByText('Happiness')).toBeDefined();
    expect(component.getByText('Welcome to Happiness')).toBeDefined();
    expect(component.getByText('Webpage Creator')).toBeDefined();
    expect(component.getByText('Our Services/Products')).toBeDefined();
  });
});
