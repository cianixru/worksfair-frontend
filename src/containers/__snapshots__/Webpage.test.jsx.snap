// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webpage.jsx Should display the Webpage Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="hero is-medium is-dark"
        >
          <div
            class="hero-body"
          >
            <div
              class="container"
            >
              <h1
                class="title"
              >
                Welcome to Happiness
              </h1>
              <h2
                class="subtitle"
              >
                Happiness truly lives here. We just happy
              </h2>
            </div>
          </div>
        </div>
        <div
          class="webpage-hero"
        >
          <div>
            <div
              class="carousel carousel-slider"
              style="width: 100%;"
            >
              <button
                class="control-arrow control-prev"
                type="button"
              />
              <div
                class="slider-wrapper axis-horizontal"
              >
                <ul
                  class="slider animated"
                  style="transform: translate3d(-50%,0,0); transition-duration: 350ms;"
                >
                  <li
                    class="slide selected"
                    style="min-width: 40%;"
                  >
                    <figure
                      class="image is-4by3"
                    >
                      <img
                        alt="sample"
                        src="https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                      />
                    </figure>
                  </li>
                  <li
                    class="slide"
                    style="min-width: 40%;"
                  >
                    <figure
                      class="image is-4by3"
                    >
                      <img
                        alt="sample"
                        src="https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                      />
                    </figure>
                  </li>
                  <li
                    class="slide selected"
                    style="min-width: 40%;"
                  >
                    <figure
                      class="image is-4by3"
                    >
                      <img
                        alt="sample"
                        src="https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                      />
                    </figure>
                  </li>
                  <li
                    class="slide"
                    style="min-width: 40%;"
                  >
                    <figure
                      class="image is-4by3"
                    >
                      <img
                        alt="sample"
                        src="https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                      />
                    </figure>
                  </li>
                </ul>
              </div>
              <button
                class="control-arrow control-next"
                type="button"
              />
              <ul
                class="control-dots"
              >
                <li
                  class="dot"
                  role="button"
                  tabindex="0"
                  value="0"
                />
                <li
                  class="dot selected"
                  role="button"
                  tabindex="0"
                  value="1"
                />
              </ul>
              <p
                class="carousel-status"
              >
                2 of 2
              </p>
            </div>
          </div>
        </div>
        <div
          class="container"
        >
          <div
            class="columns is-desktop"
          >
            <div
              class="column is-three-quarters"
            >
              <div
                class="webpage-titlebar"
                id="titlebar"
              >
                <div>
                  <h2>
                    Happiness
                     
                    <i
                      class="verified-icon"
                    />
                    <span
                      class="button is-rounded is-dark is-outlined"
                    >
                      passion
                    </span>
                  </h2>
                  <span
                    class="webpage-address"
                  >
                    <i
                      aria-hidden="true"
                      class="fa fa-map-marker"
                    />
                    No 7 Obinna Uzor Street, Canal Estate, Okota, Lagos
                  </span>
                </div>
                <div
                  class="verified-badge"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-check-circle-o"
                  />
                  Verified Business
                </div>
              </div>
            </div>
            <div
              class="column is-one-quarter"
            >
              <div
                class="card"
              >
                <header
                  class="card-header"
                >
                  <p
                    class="card-header-title"
                  >
                    Webpage Owner
                  </p>
                </header>
                <div
                  class="card-content"
                >
                  <div
                    class="media"
                  >
                    <div
                      class="media-left"
                    >
                      <figure
                        class="image is-48x48"
                      >
                        <img
                          alt="avatar"
                          class="is-rounded"
                          src="https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                        />
                      </figure>
                    </div>
                    <div
                      class="media-content content"
                    >
                      <a
                        href="/"
                      >
                        theo IO
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="hero is-medium is-dark"
      >
        <div
          class="hero-body"
        >
          <div
            class="container"
          >
            <h1
              class="title"
            >
              Welcome to Happiness
            </h1>
            <h2
              class="subtitle"
            >
              Happiness truly lives here. We just happy
            </h2>
          </div>
        </div>
      </div>
      <div
        class="webpage-hero"
      >
        <div>
          <div
            class="carousel carousel-slider"
            style="width: 100%;"
          >
            <button
              class="control-arrow control-prev"
              type="button"
            />
            <div
              class="slider-wrapper axis-horizontal"
            >
              <ul
                class="slider animated"
                style="transform: translate3d(-50%,0,0); transition-duration: 350ms;"
              >
                <li
                  class="slide selected"
                  style="min-width: 40%;"
                >
                  <figure
                    class="image is-4by3"
                  >
                    <img
                      alt="sample"
                      src="https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                    />
                  </figure>
                </li>
                <li
                  class="slide"
                  style="min-width: 40%;"
                >
                  <figure
                    class="image is-4by3"
                  >
                    <img
                      alt="sample"
                      src="https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                    />
                  </figure>
                </li>
                <li
                  class="slide selected"
                  style="min-width: 40%;"
                >
                  <figure
                    class="image is-4by3"
                  >
                    <img
                      alt="sample"
                      src="https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                    />
                  </figure>
                </li>
                <li
                  class="slide"
                  style="min-width: 40%;"
                >
                  <figure
                    class="image is-4by3"
                  >
                    <img
                      alt="sample"
                      src="https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                    />
                  </figure>
                </li>
              </ul>
            </div>
            <button
              class="control-arrow control-next"
              type="button"
            />
            <ul
              class="control-dots"
            >
              <li
                class="dot"
                role="button"
                tabindex="0"
                value="0"
              />
              <li
                class="dot selected"
                role="button"
                tabindex="0"
                value="1"
              />
            </ul>
            <p
              class="carousel-status"
            >
              2 of 2
            </p>
          </div>
        </div>
      </div>
      <div
        class="container"
      >
        <div
          class="columns is-desktop"
        >
          <div
            class="column is-three-quarters"
          >
            <div
              class="webpage-titlebar"
              id="titlebar"
            >
              <div>
                <h2>
                  Happiness
                   
                  <i
                    class="verified-icon"
                  />
                  <span
                    class="button is-rounded is-dark is-outlined"
                  >
                    passion
                  </span>
                </h2>
                <span
                  class="webpage-address"
                >
                  <i
                    aria-hidden="true"
                    class="fa fa-map-marker"
                  />
                  No 7 Obinna Uzor Street, Canal Estate, Okota, Lagos
                </span>
              </div>
              <div
                class="verified-badge"
              >
                <i
                  aria-hidden="true"
                  class="fa fa-check-circle-o"
                />
                Verified Business
              </div>
            </div>
          </div>
          <div
            class="column is-one-quarter"
          >
            <div
              class="card"
            >
              <header
                class="card-header"
              >
                <p
                  class="card-header-title"
                >
                  Webpage Owner
                </p>
              </header>
              <div
                class="card-content"
              >
                <div
                  class="media"
                >
                  <div
                    class="media-left"
                  >
                    <figure
                      class="image is-48x48"
                    >
                      <img
                        alt="avatar"
                        class="is-rounded"
                        src="https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png"
                      />
                    </figure>
                  </div>
                  <div
                    class="media-content content"
                  >
                    <a
                      href="/"
                    >
                      theo IO
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
